pkg/queue/manager.go

```diff
@@ -547,3 +552,27 @@ func (m *Manager) reportPendingWorkloads(cqName string, cq ClusterQueue) {
 	}
 	metrics.ReportPendingWorkloads(cqName, active, inadmissible)
 }
+
+func (m *Manager) GetClusterQueues() map[string]ClusterQueue {
+	m.RLock()
+	defer m.RUnlock()
+	return m.clusterQueues
```

Comment:
We should not return the map. We need to clone the keys. Returning the map means that the consumer can iterate over the map (unlocked) while another thread is modifying it. Let's have two functions: `GetClusterQueueNames()` and `GetClusterQueue(cqName string)`. Actually, `GetClusterQueue` may not be needed if we have `UpdateSnapshot(cqName) snapshot` which computes the snapshot and sets it.